### PR TITLE
[SIL] Support SILFunction as a `GraphOperationInst` attribute.

### PR DESCRIPTION
**Pull request by dan-zheng**
- Implement parsing/printing for SILFunction SymbolicValues.
  - It's necessary to include function type in the textual representation to
    enable forward references to SIL functions.
- Update test.

### CODE REVIEW

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

File: lib/SIL/SILPrinter.cpp [hunk 1]
```diff
@@ -1212,6 +1212,12 @@ class SILPrinter : public SILInstructionVisitor<SILPrinter> {
       *this << SILType::getPrimitiveObjectType(metatype.getInstanceType());
       return;
     }
+    case SymbolicValue::Function: {
+      auto function = v.getFunctionValue().first;
```
Comment (lattner) [inline lib/SIL/SILPrinter.cpp]: Nice and simple, I like it.  Please assert that the conformance in ".second" isn't present.

**Review by lattner — APPROVED** [review]

File: lib/SIL/SILPrinter.cpp (hunk 1 above)
Comment (dan-zheng) [reply lib/SIL/SILPrinter.cpp]: Done in https://github.com/apple/swift/pull/17374/commits/99ce3142f6b79bf99084768697f3f07cfe1f380b.

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

File: lib/SIL/SILPrinter.cpp [hunk 2]
```diff
@@ -1212,6 +1212,15 @@ class SILPrinter : public SILInstructionVisitor<SILPrinter> {
       *this << SILType::getPrimitiveObjectType(metatype.getInstanceType());
       return;
     }
+    case SymbolicValue::Function: {
+      assert(!v.getFunctionValue().second &&
+             "SILFunction SymbolicValues with protocol conformances cannot be "
+             "printed");
+      auto function = v.getFunctionValue().first;
```
Comment (mhong) [inline lib/SIL/SILPrinter.cpp]: auto *function

**Review by mhong — COMMENTED** [review]
Nice! Just some very minor comments.

**Pull request merged**